akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -0,0 +1,234 @@
+ * }}}
+ * or capture the result of a method in a variable:
+ * {{{
+ * var options = BackoffOptions.onFailure(childProps, childName, minBackoff, maxBackoff, randomFactor)
+ * options = options.withManualReset
+ * context.actorOf(BackoffSupervisor.props(options), name)
+ * }}}
+ */
+trait BackoffOptions {
+  /**
+   * Returns a new BackoffOptions with automatic back-off reset.
+   * The back-off algorithm is reset if the child does not crash witin the specified `resetBackoff`.
+   * @param resetBackoff The back-off is reset if the child does not crash within this duration.
+   */
+  def withAutoReset(resetBackoff: FiniteDuration): BackoffOptions
+
+  /**
+   * Returns a new BackoffOptions with manual back-off reset. The back-off is only reset
+   * if the child sends a `BackoffSupervisor.Reset` to its parent (the backoff-supervisor actor).
+   */
+  def withManualReset: BackoffOptions
+
+  /**
+   * Returns a new BackoffOptions with the supervisorStrategy.
+   * @param supervisorStrategy the supervisorStrategy that the back-off supervisor will use.
+   *   The default supervisor strategy is used as fallback if the specified supervisorStrategy (its decider)
+   *   does not explicitly handle an exception.
+   */
+  def withSupervisorStrategy(supervisorStrategy: OneForOneStrategy): BackoffOptions
+
+  /**
+   * Returns a new BackoffOptions with a default `SupervisorStrategy.stoppingStrategy`.
+   * The default supervisor strategy is used as fallback for throwables not handled by `SupervisorStrategy.stoppingStrategy`.
+   */
+  def withDefaultStoppingStrategy: BackoffOptions
+
+  /**
+   * Returns the props to create the back-off supervisor.
+   */
+  private[akka] def props: Props
+}
+
+private final case class BackoffOptionsImpl(
+  backoffType: BackoffType = RestartImpliesFailure,
+  childProps: Props,
+  childName: String,
+  minBackoff: FiniteDuration,
+  maxBackoff: FiniteDuration,
+  randomFactor: Double,
+  reset: Option[BackoffReset] = None,
+  supervisorStrategy: OneForOneStrategy = OneForOneStrategy()(SupervisorStrategy.defaultStrategy.decider)) extends BackoffOptions {
+
+  val backoffReset = reset.getOrElse(AutoReset(minBackoff))
+
+  def withAutoReset(resetBackoff: FiniteDuration) = copy(reset = Some(AutoReset(resetBackoff)))
+  def withManualReset = copy(reset = Some(ManualReset))
+  def withSupervisorStrategy(supervisorStrategy: OneForOneStrategy) = copy(supervisorStrategy = supervisorStrategy)
+  def withDefaultStoppingStrategy = copy(supervisorStrategy = OneForOneStrategy()(SupervisorStrategy.stoppingStrategy.decider))
+
+  def props = {
+    require(minBackoff > Duration.Zero, "minBackoff must be > 0")
+    require(maxBackoff >= minBackoff, "maxBackoff must be >= minBackoff")
+    require(0.0 <= randomFactor && randomFactor <= 1.0, "randomFactor must be between 0.0 and 1.0")
+    backoffReset match {
+      case AutoReset(resetBackoff) ⇒
+        require(minBackoff <= resetBackoff && resetBackoff <= maxBackoff)
+      case _ ⇒
```

Comment:
ok to ignore here, but let's add `// ignore` so that it's obvious that it's intentional